docs/docs/30-how-to-guides/20-managing-credentials.md

```diff
@@ -378,27 +378,54 @@ to the required ECR repositories.
 
 :::caution
 This method of authentication is a "lowest common denominator" approach that
-will work regardless of where Kargo is deployed. i.e. If running Kargo outside
-of EKS, this method will still work.
+will work regardless of where Kargo is deployed. i.e. if running Kargo outside EKS, this method will still work.
 
-If running Kargo within EKS, you may wish to consider using EKS Pod Identity
+If running Kargo within EKS, you may wish to either consider using EKS Pod Identity or IRSA
 instead.
 :::
 
-#### EKS Pod Identity
+#### EKS Pod Identity or IAM Roles for Service Accounts (IRSA)
 
 If Kargo locates no `Secret` resources matching a repository URL, and if Kargo
-is deployed within an EKS cluster, it will attempt to use
+is deployed within an EKS cluster, it will first attempt to use
 [EKS Pod Identity](https://docs.aws.amazon.com/eks/latest/userguide/pod-identities.html)
 to authenticate, but this relies upon some external setup. Leveraging this
 option eliminates the need to store credentials in a `Secret` resource.
 
-First, follow
+If you opt to use EKS Pod Identity, follow
 [this overview](https://docs.aws.amazon.com/eks/latest/userguide/pod-identities.html#pod-id-setup-overview)
 to set up EKS Pod Identity in your EKS cluster and assign an IAM role to the
 `kargo-controller` `ServiceAccount` within the `Namespace` to which Kargo is (or
 will be) installed.
 
+If you opt to instead use IRSA, follow [this overview](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) 
+to set up IRSA in your EKS cluster and assign an IAM role to the
+`kargo-controller` `ServiceAccount` within the `Namespace` to which Kargo is (or
+will be) installed.
+
+:::note
+In the case of IRSA you will also need to set the annotation `eks.amazonaws.com/role-arn: <iam-role-arn>` 
+to the Kargo controller service account. You can set this in the Helm chart with 
+`controller.serviceAccount.annotations`. 
+:::
+
+Regardless of the method chosen, the kargo-controller IAM role needs to be allowed to assume `kargo-project-*` roles. An
+example IAM policy doing so can be seen below. This can be scoped down to setting an explicit list of IAM role ARNs if
+required.
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": "sts:AssumeRole",
+            "Resource": "arn:aws:iam::ACCOUNT_ID:role/kargo-project-*"
+        }
+    ]
+}
```

Comment:
I know we had some disagreement over how in-depth we wanted to go on policy details. I know your concern was that not being explicit enough will result in additional support cases. I think being too prescriptive can yield similar results: Instead of hearing, "I can't figure this AWS stuff out," we'll be hearing, "I tried what the docs said and it didn't work." Worse, is someone more seasoned in IAM than we are complaining, "The way the docs say to do it isn't right," _which strikes me as a distinct possibility, because what you have above is not how I would have done it._ (Which is not to say my way is right and yours is wrong. But if there's more than one way to do it, you can be sure _someone_ with a strong opinion is going to tell you that your way is wrong.)

I guess I am invoking "less is more" here. My gut says the people who would consider using Pod Identity and/or IRSA in the first place are going to already be well-versed in IAM. This won't be their first rodeo. If we give them the few details that are actually Kargo-specific (such as the naming pattern for Project-specific roles), they'll be successful. If I'm proven wrong, we can totally revisit this.